server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -2106,6 +2109,38 @@ public void createSchemaTemplate(CreateTemplatePlan plan) throws MetadataExcepti
     }
   }
 
+  public void appendSchemaTemplate(AppendTemplatePlan plan) throws MetadataException {
+    try {
+      templateManager.appendSchemaTemplate(plan);
+      // write wal
+      if (!isRecovering) {
+        logWriter.appendSchemaTemplate(plan);
+      }
+    } catch (IOException e) {
+      throw new MetadataException(e);
+    }
+  }
+
+  public void pruneSchemaTemplate(PruneTemplatePlan plan) throws MetadataException {
+    try {
+      templateManager.pruneSchemaTemplate(plan);
+      // write wal
+      if (!isRecovering) {
+        logWriter.pruneSchemaTemplate(plan);
+      }
+    } catch (IOException e) {
+      throw new MetadataException(e);
+    }
+  }
+
+  public Template getTemplate(String templateName) throws MetadataException {
+    try {
+      return templateManager.getTemplate(templateName);
+    } catch (UndefinedTemplateException e) {
+      throw new MetadataException(e);
+    }
+  }
```

Comment:
This method maybe unnecessary. MManager could use templateManager.getTemplate directly.

Reply:
This method has been removed now. TSServiceImpl now should access Templates via MManager methods.